service/resource/vaultcrt/delete.go

```diff
@@ -46,6 +46,11 @@ func (r *Resource) ProcessDeleteState(ctx context.Context, obj, deleteState inte
 	if secretToDelete != nil {
 		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "deleting the sercet in the Kubernetes API")
 
+		ns := key.SecretNamespace(customObject)
+		if ns == "" {
+			ns = r.namespace
+		}
+
 		err = r.k8sClient.CoreV1().Secrets(r.namespace).Delete(secretToDelete.Name, &apismetav1.DeleteOptions{})
```

Comment:
Lets make sure we use the right namespace for deletion.

Reply:
Crap, yes.